corehq/apps/api/resources/v0_1.py

```diff
@@ -65,18 +71,32 @@ def _inner(req, domain, *args, **kwargs):
 
 class LoginAndDomainAuthentication(Authentication):
 
+    def __init__(self, allow_internal=False, *args, **kwargs):
+        """
+        allow_internal:
+            set this to True to allow session based access to this resource
```

Comment:
The name `allow_internal` is confusing me. I wouldn't guess that it means "allow session-based access." What do you think of `allow_session_auth` instead?